app/controllers/supervisor/comments_controller.rb

```diff
@@ -0,0 +1,35 @@
+class Supervisor::CommentsController < Supervisor::BaseController
+
+  def index
+    find_comments
+  end
+
+  def create
+    comment = Comment.new(comment_params)
+    # Now that we are saving 'checks' without comment text too, we can list them in the view with the text comments.
+    # For the user, checks with or without text are treated the same (except for the mailer).
+    if can? :supervise, comment.team
+      if comment.save
+        if comment.text.present?
```

Comment:
Still working on closing a html leak where a supervisor can create comments for non-supervised teams. If that is solved, I will solve this one as well.